presto-orc/src/main/java/com/facebook/presto/orc/StreamDescriptorBuilder.java

```diff
@@ -0,0 +1,154 @@
+        }
+        StreamProperty streamProperty = new StreamProperty(parentStreamName, type, fieldName, nestedStreamIdsBuilder.build());
+        propertiesBuilder.put(typeId, streamProperty);
+    }
+
+    public static class StreamProperty
+    {
+        private final String streamName;
+        private final OrcType orcType;
+        private final String fieldName;
+        private final List<Integer> nestedStreamIds;
+
+        public StreamProperty(String streamName, OrcType orcType, String fieldName, List<Integer> nestedStreamIds)
+        {
+            this.streamName = requireNonNull(streamName, "streamName is null");
+            this.orcType = requireNonNull(orcType, "orcType is null");
+            this.fieldName = requireNonNull(fieldName, "fieldName is null");
+            this.nestedStreamIds = ImmutableList.copyOf(requireNonNull(nestedStreamIds, "nestedStreamIds is null"));
+        }
+
+        public String getStreamName()
+        {
+            return streamName;
+        }
+
+        public OrcType getOrcType()
+        {
+            return orcType;
+        }
+
+        public String getFieldName()
+        {
+            return fieldName;
+        }
+
+        public List<Integer> getNestedStreamIds()
+        {
+            return nestedStreamIds;
+        }
+
+        @Override
+        public boolean equals(Object o)
+        {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+            StreamProperty that = (StreamProperty) o;
+            return Objects.equals(streamName, that.streamName)
+                    && Objects.equals(orcType, that.orcType)
+                    && Objects.equals(fieldName, that.fieldName)
+                    && Objects.equals(nestedStreamIds, that.nestedStreamIds);
+        }
+
+        @Override
+        public int hashCode()
+        {
+            return Objects.hash(streamName, orcType, fieldName, nestedStreamIds);
+        }
+    }
+
+    public static class AllStreams
+    {
+        private final OrcDataSource orcDataSource;
+        private final Map<Integer, StreamProperty> streamIdToProperties;
+
+        public AllStreams(OrcDataSource orcDataSource, Map<Integer, StreamProperty> streamIdToProperties)
+        {
+            this.orcDataSource = requireNonNull(orcDataSource, "orcDataSource is null");
+            this.streamIdToProperties = ImmutableMap.copyOf(requireNonNull(streamIdToProperties, "streamProperties is null"));
+        }
+
+        public OrcDataSource getOrcDataSource()
+        {
+            return orcDataSource;
+        }
+
+        public StreamProperty getStreamProperty(int streamId)
+        {
+            checkState(streamIdToProperties.containsKey(streamId), "StreamId %s is missing", streamId);
+            return streamIdToProperties.get(streamId);
```

Comment:
Do `map.get` first and check the result is not null to avoid querying the map twice.